audformat/core/database.py

```diff
@@ -433,6 +441,385 @@ def duration(file: str) -> pd.Timedelta:
+            strict: bool = False,
+            original_column_names: bool = False,
+            aggregate_function: typing.Callable[
+                [pd.Series],
+                typing.Any
+            ] = None,
+    ) -> pd.DataFrame:
+        r"""Get labels by scheme.
+
+        Return all labels
+        that match the requested ``scheme``.
+        The request can be limited to selected ``tables``
+        and/or ``splits``
+        and annotated by ``additional_schemes``
+        drawn from the whole database.
+        If ``strict`` is ``False``,
+        a scheme is defined more broadly
+        and does not only match
+        schemes of the database,
+        but also columns with the same name
+        or labels of a scheme
+        with the requested name as key.
+        If at least one returned label belongs to a segmented table,
+        the returned data frame has a segmented index.
+        An ``aggregate_function`` can be provided
+        that specifies how values are combined
+        if more than one value is returned
+        per index entry.
+
+        Args:
+            scheme: scheme ID for which labels should be returned.
+                The search for the labels
+                can be limited
+                by the ``tables`` and ``splits`` arguments
+            additional_schemes: scheme ID or sequence of scheme IDs
+                for which additional labels should be returned.
+                The search is not affected
+                by the ``tables`` and ``splits`` arguments
+            tables: limit search for ``scheme`` to selected tables
+            splits: limit search for ``scheme`` to selected splits
+            strict: if ``False``
+                not only values with the associated scheme(s)
+                are returned,
+                but also values
+                from columns with a column ID matchting the scheme
+                and values from scheme labels,
+                for matching dictionary keys
+            original_column_names: if ``True``
+                return the requested labels
+                under their original column names.
+                For mapped schemes,
+                the column name before mapping is returned,
+                e.g. when requesting ``'gender'``
+                it might return a column named ``'speaker'``
+            aggregate_function: callable to aggregate overlapping values.
+                The function gets a :class:`pandas.Series`
+                with overlapping values
+                as input.
+                E.g. set to
+                ``lambda y: y.mean()``
+                to average the values
+                or to
+                ``tuple``
+                to return them as a tuple
+
+        Returns:
+            data frame with values
+
+        Examples:
+            Return all labels that match a requested scheme.
+
+            >>> db = Database('mydb')
```

Comment:
The following examples are quite difficult to grasp and seem a bit lengthy for the Examples section. I wonder if we should replace it with a simple `emodb` example and instead move them to a dedicated Usage section that provides some more information?

Reply:
I switched to `emodb` for using the examples, but one of the examples results in an error, that I do not understand yet:

```python
import audb
import audformat

db = audb.load('emodb', version='1.4.1', only_metadata=True, full_path=False)
db['emotion']['random'] = audformat.Column(scheme_id='emotion')
db['emotion']['random'].set(db['emotion']['emotion'].get().sample(frac=1, random_state=1))
db.get('emotion', aggregate_function=lambda y: y.mode())
```
results in
```
ValueError                                Traceback (most recent call last)
Cell In[10], line 1
----> 1 db.get('emotion', aggregate_function=lambda y: y.mode())

File ~/git/audeering/audformat/audformat/core/database.py:805, in Database.get(self, scheme, additional_schemes, tables, splits, strict, original_column_names, aggregate_function)
    798     ys_requested_scheme = [
    799         y.astype(data.dtype)
    800         for y in ys_requested_scheme
    801     ]
    803 index = utils.union([y.index for y in ys_requested_scheme])
--> 805 obj = utils.concat(
    806     ys_requested_scheme,
    807     aggregate_function=aggregate_function,
    808 )
    809 obj = obj.loc[index]
    811 if len(obj) == 0:

File ~/git/audeering/audformat/audformat/core/utils.py:361, in concat(objs, overwrite, aggregate_function)
    359         print(f'{y=}')
    360         print(f'{y.index=}')
--> 361         columns_reindex[column].loc[y.index] = y
    363 # Use `None` to force `{}` return the correct index, see
    364 # https://github.com/pandas-dev/pandas/issues/52404
    365 df = pd.DataFrame(columns_reindex or None, index=index)

File ~/.envs/analysis/lib/python3.8/site-packages/pandas/core/indexing.py:818, in _LocationIndexer.__setitem__(self, key, value)
    815 self._has_valid_setitem_indexer(key)
    817 iloc = self if self.name == "iloc" else self.obj.iloc
--> 818 iloc._setitem_with_indexer(indexer, value, self.name)

File ~/.envs/analysis/lib/python3.8/site-packages/pandas/core/indexing.py:1797, in _iLocIndexer._setitem_with_indexer(self, indexer, value, name)
   1795     self._setitem_with_indexer_split_path(indexer, value, name)
   1796 else:
-> 1797     self._setitem_single_block(indexer, value, name)

File ~/.envs/analysis/lib/python3.8/site-packages/pandas/core/indexing.py:2072, in _iLocIndexer._setitem_single_block(self, indexer, value, name)
   2069     value = self._align_series(indexer, Series(value))
   2071 elif isinstance(value, ABCDataFrame) and name != "iloc":
-> 2072     value = self._align_frame(indexer, value)
   2074 # check for chained assignment
   2075 self.obj._check_is_chained_assignment_possible()

File ~/.envs/analysis/lib/python3.8/site-packages/pandas/core/indexing.py:2358, in _iLocIndexer._align_frame(self, indexer, df)
   2355         val = df.reindex(index=ax)._values
   2356     return val
-> 2358 raise ValueError("Incompatible indexer with DataFrame")

ValueError: Incompatible indexer with DataFrame
```

Don't know yet if this is related to the `pandas` issue reported at https://github.com/pandas-dev/pandas/issues/47229 or if there is something we need to fix inside `audformat.utils.concat()`.

Reply:
Sorry, was my fault as `lambda y: y.mode()` was not returning the correct structure.

I updated now all examples to use `emodb`, see also the updated screenshots in the description.